internal/internal_task_handlers.go

```diff
@@ -292,21 +293,21 @@ func isCommandEvent(eventType enumspb.EventType) bool {
 
 // NextCommandEvents returns events that there processed as new by the next command.
 // TODO(maxim): Refactor to return a struct instead of multiple parameters
-func (eh *history) NextCommandEvents() (result []*historypb.HistoryEvent, markers []*historypb.HistoryEvent, binaryChecksum string, sdkFlags []sdkFlag, err error) {
+func (eh *history) NextCommandEvents() (result []*historypb.HistoryEvent, markers []*historypb.HistoryEvent, binaryChecksum string, sdkFlags []sdkFlag, msgs []*protocolpb.Message, err error) {
```

Comment:
I think it's probably time to address that TODO. These functions return a hilarious amount of things

Reply:
I agree we need to refactor this. I left it out in this PR because this code is brittle and I'd prefer to the a refactor where we fix all the issues in this function including potential pagination issues and refactor to use a struct.

Reply:
I kinda want to see how far we can go with this.